app/gui/view/graph-editor/src/component/node/input/widget.rs

```diff
@@ -104,72 +311,74 @@ pub mod dot {
-#[derive(Clone, Debug, CloneRef)]
-pub struct SingleChoice {
-    display_object: display::object::Instance,
-    frp:            Frp,
+#[derive(Debug)]
+pub struct SingleChoiceModel {
     #[allow(dead_code)]
+    network:        frp::Network,
     dropdown:       Rc<RefCell<LazyDropdown>>,
     /// temporary click handling
     activation_dot: dot::View,
 }
 
-impl SingleChoice {
-    fn new(app: &Application, argument_info: span_tree::ArgumentInfo, node_height: f32) -> Self {
-        let display_object = display::object::Instance::new();
-        let dropdown = app.new_view::<Dropdown<ImString>>();
-        display_object.add_child(&dropdown);
-        let frp = Frp::new();
-        let network = &frp.network;
-        let input = &frp.input;
-        dropdown.set_y(-node_height);
-        dropdown.set_max_open_size(Vector2(300.0, 500.0));
+impl SingleChoiceModel {
+    fn new(common: &ModelCommon, frp: &SampledFrp) -> Self {
+        let display_object = &common.display_object;
 
         let activation_dot = dot::View::new();
-        let color: color::Rgba = color::Lcha::new(0.56708, 0.23249, 0.71372, 1.0).into();
-        activation_dot.color.set(color.into());
         activation_dot.set_size((15.0, 15.0));
-        activation_dot.set_y(-node_height / 2.0);
         display_object.add_child(&activation_dot);
 
-
-        let dropdown = if !argument_info.tag_values.is_empty() {
-            let entries: Vec<ImString> = argument_info.tag_values.iter().map(Into::into).collect();
-            let app = app.clone_ref();
-            let display_object = display_object.clone_ref();
-            let frp = frp.clone_ref();
-            // Register a watcher for current value, to allow its value to be retrieved at any time
-            // when dropdown is initialized.
-            let current_value_watch = input.set_current_value.register_watch();
-            let lazy = LazyDropdown::NotInitialized {
-                app,
-                display_object,
-                frp,
-                node_height,
-                entries,
-                _current_value: current_value_watch,
-            };
-            Rc::new(RefCell::new(lazy))
-        } else {
-            // TODO [PG]: Support dynamic entries.
-            // https://www.pivotaltracker.com/story/show/184012743
-            unimplemented!();
+        frp::new_network! { network
+            init <- source_();
+            let focus_in = display_object.on_event::<event::FocusIn>();
+            let focus_out = display_object.on_event::<event::FocusOut>();
+            is_focused <- bool(&focus_out, &focus_in);
+            is_open <- frp.set_visible && is_focused;
+            is_open <- is_open.sampler();
         };
 
+        let set_current_value = frp.set_current_value.clone_ref();
+        let dropdown_output = frp.out_value_changed.clone_ref();
+        let dropdown = LazyDropdown::new(
+            &common.app,
```

Comment:
why is it called "common"? I did not find this in other places of our codebase and its not obvious to me why this name was chosen.